### PR TITLE
Include leaf in SchemaBuilder by default

### PR DESCRIPTION
**Pull request by CraigMacomber**
## Description

Refactor SchemaBuilder to move more of its logic into the base class, then move it to `domains` where its allowed to depend on `leaf`.

To reduce cases where having leaf types in the schema cause issues, contextuallyTyped.ts was updated to disambiguate more cases related to leaf nodes: for example an untyped object is now checked for valueSymbol, and it not present, the object is considered unambiguously not a leaf.

## Breaking Changes

The identities for the Json domain now are in the `com.fluidframework.json` scope and lowercase.
The leaf domain is included by default when using SchemaBuilder (this can add more allowed types to `Any`, and impact compatibility of existing document when schematizing)

## Reviewer Guidance

The review process is outlined on [this wiki page](https://github.com/microsoft/FluidFramework/wiki/PR-Guidelines#guidelines).

### CODE REVIEW

File: api-report/tree2.api.md [hunk 1]
```diff
@@ -1166,25 +1162,25 @@ export interface JsonableTree extends GenericTreeNode<JsonableTree> {
...
 }>]>;
-}>, () => TreeSchema<"Json.Array", any>, TreeSchema<"com.fluidframework.leaf.number", {
+}>, () => TreeSchema<"com.fluidframework.json.Array", any>, TreeSchema<"com.fluidframework.leaf.number", {
 leafValue: import("../..").ValueSchema.Number;
 }>, TreeSchema<"com.fluidframework.leaf.boolean", {
 leafValue: import("../..").ValueSchema.Boolean;
 }>, TreeSchema<"com.fluidframework.leaf.string", {
 leafValue: import("../..").ValueSchema.String;
-}>, TreeSchema<"Json.Null", {
+}>, TreeSchema<"com.fluidframework.json.Null", {
```
Comment (DLehenbauer) [inline api-report/tree2.api.md:1183]: Consider the convention of using lowercase for reverse-domain based ids?

Motivation 1: Differentiating types based on case is an anti-pattern (e.g., TS 'object' vs. 'Object')
Motivation 2: Capitalizing Null "looks wrong" to me.

File: experimental/dds/tree2/src/feature-libraries/schemaBuilderBase.ts [hunk 2]
```diff
@@ -16,9 +17,44 @@ import {
...
+	 */
+	name?: string;
+
+	/**
+	 * Optional configuration for "linting". See {@link SchemaLintConfiguration}. Currently defaults to enabling all lints.
+	 */
+	lint?: Partial<SchemaLintConfiguration>;
+
+	/**
+	 * Libraries to include in this one.
```
Comment (Josmithr) [inline experimental/dds/tree2/src/feature-libraries/schemaBuilderBase.ts:46]: Is it worth noting that the leaf domain is always included?

File: experimental/dds/tree2/src/feature-libraries/schemaBuilderBase.ts [hunk 3]
```diff
@@ -16,9 +17,44 @@ import {
...
+	 *
+	 * Unlike adding of individual schema, adding of libraries is idempotent.
```
Comment (Josmithr) [inline experimental/dds/tree2/src/feature-libraries/schemaBuilderBase.ts:48]: Nit
```suggestion
	 * @remarks
	 * Unlike adding of individual schema, adding of libraries is idempotent.
```

File: experimental/dds/tree2/src/feature-libraries/schemaBuilderBase.ts (hunk 2 above)
Comment (CraigMacomber) [reply experimental/dds/tree2/src/feature-libraries/schemaBuilderBase.ts:46]: It is not included in this base class. Doing so would be invalid (a cycle) as the leaf domain is implemented using this class.

File: experimental/dds/tree2/src/feature-libraries/schemaBuilderBase.ts [hunk 4]
```diff
@@ -179,6 +358,12 @@ export class SchemaBuilderBase<TScope extends string, TName extends number | str
 	>(kind: Kind, ...allowedTypes: T): FieldSchema<Kind, T> {
 		return FieldSchema.createUnsafe(kind, allowedTypes);
 	}
+
+	protected normalizeField<TSchema extends ImplicitFieldSchema>(
```
Comment (Josmithr) [inline experimental/dds/tree2/src/feature-libraries/schemaBuilderBase.ts:362]: Docs? 😋

File: experimental/dds/tree2/src/domains/schemaBuilder.ts [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+
+	/**
+	 * Define a schema for a {@link RequiredField}.
+	 * @remarks
+	 * Shorthand or passing `FieldKinds.required` to {@link FieldSchema.create}.
+	 * Note that `FieldKinds.required` is the current default field kind, so APIs accepting {@link ImplicitFieldSchema}
+	 * can be passed the `allowedTypes` and will implicitly wrap it up in a {@link RequiredField}.
+	 *
+	 * Since this creates a {@link FieldSchema} (and not a {@link TreeSchema}), the resulting schema is structurally typed, and not impacted by the {@link SchemaBuilderBase.scope}:
+	 * therefor this method is the same as the static version.
```
Comment (noencke) [inline experimental/dds/tree2/src/domains/schemaBuilder.ts:85]:
```suggestion
	 * therefore this method is the same as the static version.
```

File: experimental/dds/tree2/src/domains/schemaBuilder.ts [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+	 */
+	public static sequence = fieldHelper(FieldKinds.sequence);
+
+	/**
+	 * Define a schema for a {@link Sequence}.
+	 * @remarks
+	 * Shorthand or passing `FieldKinds.sequence` to {@link FieldSchema.create}.
+	 *
+	 * Since this creates a {@link FieldSchema} (and not a {@link TreeSchema}), the resulting schema is structurally typed, and not impacted by the {@link SchemaBuilderBase.scope}:
+	 * therefor this method is the same as the static version.
```
Comment (noencke) [inline experimental/dds/tree2/src/domains/schemaBuilder.ts:104]:
```suggestion
	 * therefore this method is the same as the static version.
```

**Review by noencke — APPROVED** [review]

File: api-report/tree2.api.md (hunk 1 above)
Comment (CraigMacomber) [reply api-report/tree2.api.md:1183]: done

**Review by noencke — APPROVED** [review]

**Pull request merged**